src/SerdeSpecialValue.php

```diff
@@ -0,0 +1,10 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Crell\Serde;
+
+enum SerdeSpecialValue
```

Comment:
Naming: One of the two hardest problems in CS...

"Special Value" seems lazy to me.  We're really using it as a sentinel value.  So, perhaps `SerdeSentinel`?